lib/ember-orbit/store.js

```diff
@@ -1,379 +1,145 @@
-  },
-
-  settleTransforms: function() {
-    return this.orbitSource.settleTransforms();
-  },
-
-  willDestroy: function() {
-    this.orbitSource.off('didTransform', this.didTransform, this);
-    this._recordArrayManager.destroy();
-    this._super.apply(this, arguments);
-  },
-
-  typeMapFor: function(type) {
-    var typeMap = this.typeMaps[type];
-
-    if (typeMap) return typeMap;
-
-    typeMap = {
-      records: {},
-      type: type
-    };
-
-    this.typeMaps[type] = typeMap;
-
-    return typeMap;
-  },
-
-  transform: function(operation) {
-    return this.orbitSource.transform(operation);
-  },
-
-  all: function(type) {
-    this._verifyType(type);
-
-    var typeMap = this.typeMapFor(type),
-        findAllCache = typeMap.findAllCache;
-
-    if (findAllCache) { return findAllCache; }
-
-    var array = this._recordArrayManager.createRecordArray(type);
-
-    typeMap.findAllCache = array;
-    return array;
-  },
-
-  filter: function(type, query, filter) {
-    this._verifyType(type);
-
-    var length = arguments.length;
-    var hasQuery = length === 3;
-    var promise;
-    var array;
-
-    if (hasQuery) {
-      promise = this.query(type, query);
-    } else if (length === 2) {
-      filter = query;
-    }
-
-    if (hasQuery) {
-      array = this._recordArrayManager.createFilteredRecordArray(type, filter, query);
-    } else {
-      array = this._recordArrayManager.createFilteredRecordArray(type, filter);
-    }
-
-    promise = promise || RSVP.Promise.cast(array);
-
-    return promiseArray(promise.then(function() {
-      return array;
-    }, null, "OE: Store#filter of " + type));
-  },
-
-  find: function(type, id, options) {
-    var _this = this;
-    this._verifyType(type);
-
-    var promise = this.orbitSource.find(type, id, options).then(function(data) {
-      return _this._lookupFromData(type, data);
-    });
-
-    return this._request(promise);
-  },
-
-  query: function(type, query, options) {
-    var _this = this;
-    this._verifyType(type);
-
-    var promise = this.orbitSource.query(type, query, options).then(function(data) {
-      return _this._lookupFromData(type, data);
-    });
-
-    return this._request(promise);
-  },
-
-  add: function(type, properties) {
-    var _this = this;
-    this._verifyType(type);
-    properties = properties || {};
-
-    get(this, 'schema').normalize(type, properties);
-    var promise = this.orbitSource.add(type, properties).then(function(data) {
-      return _this._lookupFromData(type, data);
-    });
-
-    return this._request(promise);
-  },
+    const orbitCache = orbitStore.cache;
 
-  remove: function(type, id) {
-    this._verifyType(type);
-    id = this._normalizeId(id);
+    const schema = Schema.create({ container, _orbitSchema: orbitStore.schema });
+    this._identityMap = IdentityMap.create({ _schema: schema, _orbitCache: orbitCache, _store: this });
+    const cache = Cache.create({ _orbitCache: orbitCache, _identityMap: this._identityMap });
```

Comment:
I found the existing implementation was a bit of a chore to configure in the initializers so I've taken the approach of registering a configured orbitStore in the registry. I'm keen to have a look at options for configuring orbit easier though, presets etc.